Rename a's keys as local's keys and b's keys as remote's keys

ariard opened the pull request:
Downstreaming this while working on anchor, follow-up of #613. This is an attempt to make keys variable names less confusing based on script semantic as an invariant.

Variables should be named according to the script semantic which is
an invariant with regards to generating a local or remote commitment
transaction.

I.e a local_delayed_payment key will always offer to the owner
of the commitment transaction to be paid back after a given delay.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -374,7 +374,7 @@ impl_writeable!(HTLCOutputInCommitment, 1 + 8 + 4 + 32 + 5, {
 
 #[inline]
-pub(crate) fn get_htlc_redeemscript_with_explicit_keys(htlc: &HTLCOutputInCommitment, a_htlc_key: &PublicKey, b_htlc_key: &PublicKey, revocation_key: &PublicKey) -> Script {
+pub(crate) fn get_htlc_redeemscript_with_explicit_keys(htlc: &HTLCOutputInCommitment, local_htlc_key: &PublicKey, remote_htlc_key: &PublicKey, revocation_key: &PublicKey) -> Script {
```
This isnt always the remote/local key though, no? We also call this with the keys swapped to get redeemscripts for *their* transactions. We probably need to standardize on some kind of nomenclature cause remote/local is often confusing.

ariard replied:
From the script layout as described in the BOLT3, how `local_htlcpubkey` is instanced depends if we are building a transaction for _us _ or for our counterparty.

I think when we _generate_ a transaction, including its scripts, there is no such thing as`local` or `remote` it's always a local commitment transaction. Now when we _sign_ such transaction and what we sign depends if its a local transaction or remote one.

So, as a nomenclature,  I propose:
* for all our helpers in `chan_utils` bind to local/remote to match BOLT3
* in `channel/channelmonitor` label transaction as local_tx or remote_tx (no changes expected)
* for holding our or counterparty elements such as `their_keys` in `Channel` bind to owner/counterparty avoid confusion when we instance our helpers.

We may have to move helpers from `Channel` to `chan_utils` to make it cleaner.

jkczyz replied:
The naming is definitely contextual. So I would agree that we should not settle on always one way of describing the relationship.

I would like to see "counterparty" used when generally referring to the other side (i.e., when the relationship can be flipped). I don't have a strong opinion on "owner" being the opposite of "counterparty" just yet. It may make sense to not have any prefix in that case. I'd need to see a concrete example.

I'm generally not in favor of using possessive pronouns like "our" and "their" in code. I'd prefer a more descriptive name for the relationship being modeled.

ariard replied:
@jkczyz fyi, even the BOLT isn't clear on this https://github.com/lightningnetwork/lightning-rfc/pull/638

> The naming is definitely contextual. So I would agree that we should not settle on always one way of describing the relationship.

I agree, do you see another context that the ones I'm describing, namely `chan_utils`, `Channel`, `channel/channelmonitor` or if those contexts make sense ? Maybe 2) and 3) are the same.

jkczyz replied:
My thinking was that if `chan_utils` is suppose to be agnostic to what is local and what is remote (at least in some places), then the "local" and "remote" naming may not make sense in those particular contexts.

So for instance, AFAICT, `TxCreationKeys` should not use the words "local" and "remote" since it is used in both contexts:

https://github.com/rust-bitcoin/rust-lightning/blob/c09c9fb9a00ebaf60cd44755e7a14df1dbb7c7a7/lightning/src/ln/channel.rs#L1102

https://github.com/rust-bitcoin/rust-lightning/blob/c09c9fb9a00ebaf60cd44755e7a14df1dbb7c7a7/lightning/src/ln/channel.rs#L1116

Additionally, there is a lot of redundancy in some of the naming that isn't necessary in the given context. e.g., `initial_local_commitment_tx.local_keys.local_htlc_key`.

TheBlueMatt replied:
Right there's kind of two different things we'll want two words for:
 * Something is our's/our counterparty's (I think I like ours/counterpartys or so?)
 * Something is for the "owner of the transaction", which is what local/remote mean here.

Its true the BOLTs are a mess in this regard, but thats precisely why we should try to do better.

As a concrete proposal, I think ours/counterpartys and local/remote (which is whats there already in TxCreationKeys) is reasonable, but am open to other ideas.

jkczyz replied:
For `ours` and `counterpartys`, I would prefer no prefix and `counterparty`.

ariard commented:
Updated with a new commit re-labeling fields according to Matt's suggestion: https://github.com/rust-bitcoin/rust-lightning/pull/633#discussion_r436307759. "Local"/"Remote" references aren't all dead in channel.rs but it kill a lot of them. 

Lmk if you find this clearer and worthy of the change.

ariard commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -317,13 +317,13 @@ impl_writeable!(ChannelPublicKeys, 33*5, {
-			a_htlc_key: derive_public_key(&secp_ctx, &per_commitment_point, &a_htlc_base)?,
-			b_htlc_key: derive_public_key(&secp_ctx, &per_commitment_point, &b_htlc_base)?,
-			a_delayed_payment_key: derive_public_key(&secp_ctx, &per_commitment_point, &a_delayed_payment_base)?,
+			revocation_key: derive_public_revocation_key(&secp_ctx, &per_commitment_point, &remote_revocation_base)?,
```
We never talk about it, no more we talk about `remote_delayedpubkey` in script format. If we drop, by consistency we should also drop the `local_` prefix for `local_delayed_payment_base`. Seems a shortcoming of the spec here.

jkczyz replied:
Not sure that I follow. What is a shortcoming of the spec? Trying to learn more than anything.

ariard replied:
For e.g, if you take offered script with a `revocationpubkey`. It should have been called `remote_revocationpubkey` even if there is a contribution from local (`per_commitment_point`) it's used by remote to punish you. Keys names should be labeled according to their ownership.

ariard commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -627,9 +627,9 @@ impl LocalCommitmentTransaction {
+				let htlc_tx = build_htlc_transaction(&txid, self.feerate_per_kw, local_csv, &this_htlc.0, &self.local_keys.local_delayed_payment_key, &self.local_keys.revocation_key);
 
-				let htlc_redeemscript = get_htlc_redeemscript_with_explicit_keys(&this_htlc.0, &self.local_keys.a_htlc_key, &self.local_keys.b_htlc_key, &self.local_keys.revocation_key);
+				let htlc_redeemscript = get_htlc_redeemscript_with_explicit_keys(&this_htlc.0, &self.local_keys.local_htlc_key, &self.local_keys.remote_htlc_key, &self.local_keys.revocation_key);
```
Dropped prefix `local_` for keys.

ariard commented on lightning/src/ln/channel.rs:
```diff
@@ -1893,13 +1893,13 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		}
 
-		let their_funding_pubkey = self.their_pubkeys.as_ref().unwrap().funding_pubkey;
+		let funding_pubkey = self.counterparty_pubkeys.as_ref().unwrap().funding_pubkey;
```
You're right, I've been confused by this first one. In fact the pubkey has a clear owner but redeemScript and txo is common. Corrected.

jkczyz commented on lightning/src/ln/channelmanager.rs:
```diff
@@ -2211,20 +2211,20 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 	}
 
-	fn internal_open_channel(&self, their_node_id: &PublicKey, their_features: InitFeatures, msg: &msgs::OpenChannel) -> Result<(), MsgHandleErrInternal> {
+	fn internal_open_channel(&self, counterparty_node_id: &PublicKey, their_features: InitFeatures, msg: &msgs::OpenChannel) -> Result<(), MsgHandleErrInternal> {
```
Personally, for all these methods I'd just call the parameter `node_id` and `features`. From the context, it should be clear that we aren't passing in our own node id and features.

ariard replied:
I'm ~0 on this. It's quite obvious when you're used to, but for newcomers that's clearly helpful ?

jkczyz replied:
Not sure what ~0 means. 😄  Indifferent?

My argument is these are (helpers for) implementing `ChannelMessageHandler`, so it should be clear. I'd be fine with leaving these unchanged and moving to `node_id` and `features` in a follow-up that included updating `ChannelMessageHandler`.

ariard replied:
Yes indifferent, let's keep it for a follow-up.

ariard commented on lightning/src/ln/channel.rs:
```diff
@@ -1447,27 +1447,27 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-		let local_initial_commitment_tx = self.build_commitment_transaction(self.cur_local_commitment_transaction_number, &local_keys, true, false, self.feerate_per_kw, logger).0;
-		let local_sighash = hash_to_message!(&bip143::SighashComponents::new(&local_initial_commitment_tx).sighash_all(&local_initial_commitment_tx.input[0], &funding_script, self.channel_value_satoshis)[..]);
+		let keys = self.build_local_transaction_keys(self.cur_commtment_transaction_number)?;
+		let initial_commitment_tx = self.build_commitment_transaction(self.cur_commtment_transaction_number, &keys, true, false, self.feerate_per_kw, logger).0;
```
I interpret code here as "We build a commitment transaction. Owner is us" So no prefix.

jkczyz replied:
Hmm, so I thought this would have fallen under the second bullet in the linked comment. Would we ever say `local_commitment_tx` and `remote_commitment_tx` in your interpretation? If so, in what context?

ariard replied:
I think we should restrain pointed distinction (local/remote) to templates as used by our chan helpers (`chan_utils.rs`) and avoid leaking local/remote in Channel storage fields. Even it's not perfect for now.

ariard commented on lightning/src/ln/channel.rs:
```diff
@@ -1447,27 +1447,27 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-		let remote_initial_commitment_tx = self.build_commitment_transaction(self.cur_remote_commitment_transaction_number, &remote_keys, false, false, self.feerate_per_kw, logger).0;
-		let remote_signature = self.local_keys.sign_remote_commitment(self.feerate_per_kw, &remote_initial_commitment_tx, &remote_keys, &Vec::new(), self.our_to_self_delay, &self.secp_ctx)
+		let counterparty_keys = self.build_remote_transaction_keys()?;
+		let counterparty_initial_commitment_tx = self.build_commitment_transaction(self.cur_counterparty_commitment_transaction_number, &counterparty_keys, false, false, self.feerate_per_kw, logger).0;
```
Same reasoning. Builder has the notion of local/remote, its ouput has clearly an owner.

ariard commented on lightning/src/ln/channel.rs:
```diff
@@ -4369,14 +4369,14 @@ mod tests {
+		let node_a_counterparty_node_id = PublicKey::from_secret_key(&secp_ctx, &SecretKey::from_slice(&[42; 32]).unwrap());
 		let config = UserConfig::default();
-		let mut node_a_chan = Channel::<EnforcingChannelKeys>::new_outbound(&&feeest, &&keys_provider, node_a_node_id, 10000000, 100000, 42, &config).unwrap();
+		let mut node_a_chan = Channel::<EnforcingChannelKeys>::new_outbound(&&feeest, &&keys_provider, node_a_counterparty_node_id, 10000000, 100000, 42, &config).unwrap();
 
 		// Create Node B's channel by receiving Node A's open_channel message
 		let open_channel_msg = node_a_chan.get_open_channel(genesis_block(network).header.bitcoin_hash(), &&feeest);
-		let node_b_node_id = PublicKey::from_secret_key(&secp_ctx, &SecretKey::from_slice(&[7; 32]).unwrap());
-		let mut node_b_chan = Channel::<EnforcingChannelKeys>::new_from_req(&&feeest, &&keys_provider, node_b_node_id, InitFeatures::known(), &open_channel_msg, 7, &config).unwrap();
+		let node_b_counterparty_node_id = PublicKey::from_secret_key(&secp_ctx, &SecretKey::from_slice(&[7; 32]).unwrap());
+		let mut node_b_chan = Channel::<EnforcingChannelKeys>::new_from_req(&&feeest, &&keys_provider, node_b_counterparty_node_id, InitFeatures::known(), &open_channel_msg, 7, &config).unwrap();
```
Right I think here that's zeal. Corrected.

jkczyz replied:
I believe the node ids aren't accurately named even with your change reverted. That is, `node_a_chan` should be created with node B's id. In practice, it doesn't matter since the ids aren't used after the `Channel` is created.

ariard replied:
You right, updated comments to make it clearer.

ariard commented:
Thanks @jkczyz for review, rebased and updated. 

> Thanks for making the change! Eventually, we may want to encapsulate these into separate structs with related elements (e.g., limits, balance, scripts, etc?) and expose behaviors rather than fields.

Yes, let's do this post-anchor output implementation ? I'm modifying few things here. 
I generally agree on encapsulating related elements, also to split `build_commitment_transactions` right now it's melting balance computation and pure transaction generation.

TheBlueMatt commented:
This needs rebase. Maybe instead of `local_` and `remote_` we could use `broadcaster_` and `broadcastee_` or something to describe whats really happening? local and remote are still sooo ripe for confusion.

ariard commented:
@TheBlueMatt @jkczyz Sorry for latency, finally back working on this :) 

> This needs rebase. Maybe instead of local_ and remote_ we could use broadcaster_ and broadcastee_ or something to describe whats really happening? local and remote are still sooo ripe for confusion.

I agree that local_/remote_ should disappear, I'm thinking about owner_/contributor_ as when generating a transaction it must have a unique final owner while we might integrate contributions from another participant.  These contributions can be either our key material or our counterparty's one according to message processing flow. IMO, better than broadcaster_/broadcastee_, we don't necessary broadcast commitment transactions.

Overall, there is a huge part of the codebase still bearing the local_/remote_ nomenclature, this PR is just a step forward and should have follow-ups.

TheBlueMatt commented:
> I'm thinking about owner_/contributor_

Hmm, I still think its possible to confuse that - the node which sends all the data on the wire here is not the owner, but it feels like they should be the owner since they decided when to create the commitment tx and when to send information about it (though not when to finalize it or broadcast it).

I agree broadcaster is confusing because we often don't broadcast, but I'm not sure how else to describe a potentially-broadcastable transaction. Maybe broadcastable_node_*, though that seems verbose.

ariard commented:
> Hmm, I still think its possible to confuse that - the node which sends all the data on the wire here is not the owner, but it feels like they should be the owner since they decided when to create the commitment tx and when to send information about it (though not when to finalize it or broadcast it).

I don't agree on when to create the commitment as Alice can send a `update_add_htlc/commitment_signed` to Bob, whom in reply will send back signatures for updating Alice's commitment. So you can analyze as Alice who decided the creation, assuming honest protocol behavior from Bob. 

I like owner as ultimately you're deciding for the finalization and thus authorizing this transaction to spend the funding output. Also it underscores you have to reject counterparty contribution if they're not valid (like invalid signatures/pubkeys).

Sounds we won't agree here, @jkczyz do you want serve as a tie-breaker ? Does owner/contributor sound like a meaningful improvement on local/remote :) ?

jkczyz commented on lightning/src/ln/channel.rs:
```diff
@@ -658,24 +658,24 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-		if remote_channel_reserve_satoshis < our_dust_limit_satoshis {
-			return Err(ChannelError::Close(format!("Suitable channel reserve not found. remote_channel_reserve was ({}). our_dust_limit_satoshis is ({}).", remote_channel_reserve_satoshis, our_dust_limit_satoshis)));
+		if remote_channel_reserve_satoshis < dust_limit_satoshis {
+			return Err(ChannelError::Close(format!("Suitable channel reserve not found. remote_channel_reserve was ({}). our_dust_limit_satoshis is ({}).", remote_channel_reserve_satoshis, dust_limit_satoshis)));
```
s/our_dust_limit_satoshis/dust_limit_satoshis

jkczyz commented on lightning/src/ln/channel.rs:
```diff
@@ -1385,8 +1385,8 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-		if msg.channel_reserve_satoshis < self.our_dust_limit_satoshis {
-			return Err(ChannelError::Close(format!("Peer never wants payout outputs? channel_reserve_satoshis was ({}). our_dust_limit is ({})", msg.channel_reserve_satoshis, self.our_dust_limit_satoshis)));
+		if msg.channel_reserve_satoshis < self.dust_limit_satoshis {
+			return Err(ChannelError::Close(format!("Peer never wants payout outputs? channel_reserve_satoshis was ({}). our_dust_limit is ({})", msg.channel_reserve_satoshis, self.dust_limit_satoshis)));
```
s/our_dust_limit/dust_limit_satoshis

jkczyz commented on lightning/src/ln/channel.rs:
```diff
@@ -1865,13 +1865,12 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-			if self.value_to_self_msat < self.local_channel_reserve_satoshis * 1000 + local_commit_tx_fee_msat {
-				return Err(ChannelError::Close("Cannot receive value that would put us under local channel reserve value".to_owned()));
+			if self.value_to_self_msat < self.channel_reserve_satoshis * 1000 + local_commit_tx_fee_msat {
+				return Err(ChannelError::Close("Cannot receive value that would put us under our channel reserve value".to_owned()));
```
The side ("our") of the reserve is immaterial here since its implied by this being a receive. I'd try to avoid "us" and "our" in these messages by stating as:

> Cannot receive value that would put the channel under its reserve value

That said, there are other messages that use "our" and "their" pronouns it appears.

ariard replied:
Good suggestion but I took it a bit differently "Cannot accept HTLC that would put counterparty balance under channel reserve value" as it underscores both purpose of this check and ownership of checked value.

jkczyz commented on lightning/src/ln/channel.rs:
```diff
@@ -2804,22 +2803,22 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		// AwaitingRemoteRevoke set, which indicates we sent a commitment_signed but haven't gotten
 		// the corresponding revoke_and_ack back yet.
-		let our_next_remote_commitment_number = INITIAL_COMMITMENT_NUMBER - self.cur_remote_commitment_transaction_number + if (self.channel_state & ChannelState::AwaitingRemoteRevoke as u32) != 0 { 1 } else { 0 };
+		let next_remote_commitment_number = INITIAL_COMMITMENT_NUMBER - self.cur_counterparty_commitment_transaction_number + if (self.channel_state & ChannelState::AwaitingRemoteRevoke as u32) != 0 { 1 } else { 0 };
```
I guess this where I'm a bit confused. Here we are mixing "counterparty" and "remote".

ariard replied:
My bad, I think this variable was really badly-named. Switched it to `next_counterparty_commitment_number` as we used this value to build counterparty's commitment transaction.

jkczyz commented on lightning/src/ln/channel.rs:
```diff
@@ -3746,18 +3745,18 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-			return Err(ChannelError::Ignore(format!("Cannot send value that would put us under local channel reserve value ({})", chan_reserve_msat)));
+		let chan_reserve_msat = self.channel_reserve_satoshis * 1000;
+		if pending_value_to_self_msat - amount_msat - commit_tx_fee_msat < chan_reserve_msat {
+			return Err(ChannelError::Ignore(format!("Cannot send value that would put us under our channel reserve value ({})", chan_reserve_msat)));
```
Likewise.

jkczyz commented:
> Sounds we won't agree here, @jkczyz do you want serve as a tie-breaker ? Does owner/contributor sound like a meaningful improvement on local/remote :) ?

I think I got a little lost in the discussion. :) Could you point me to a concrete example in the PR where these would be used in replace of local/remote?

ariard commented:
Thanks @jkczyz for the new parse. As an example of new `owner/contributor` usage, see `TxCreationKeys` in https://github.com/rust-bitcoin/rust-lightning/pull/633/commits/2fb8a217c3898a9cf877238c3eb1b777f0d68e56 

@TheBlueMatt what about `remittee/recipient` instead of `owner` ? Transactions are like letters, issued and signed by a sender but aiming to an exclusive remitte to whom only they will provide value.

TheBlueMatt commented:
`remittee/recipient` or `remittee/countersignatory` sounds maybe ok for payments, as they are remitted, but I don't think you'd generally use the term "remit" for pure data - personally I'm still a fan of `broadcastor`-based language as the whole point of a transaction is to possibly be broadcast, even if it never is.

ariard commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -237,12 +237,12 @@ pub fn derive_private_revocation_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1
-	part_b.mul_assign(&commit_append_rev_hash_key)?;
-	part_a.add_assign(&part_b[..])?;
-	Ok(part_a)
+	let mut counterparty_contrib = revocation_base_secret.clone();
```
That's intentional, as we never derive counterparty revocation key as it's only used to punish them, revocation appears to me directed thus the counterparty_ nomenclature. I concede that can be confusing ?

TheBlueMatt replied:
Its also a pub function, so users can do whatever :p. But, sure, that's fine with me, maybe leave a comment, though.

ariard commented:
Updated with restoring the `local_` prefix for our to_self_delay in keysinterface, if I understood well your concern. Even if we have context there to dissociate, I concede that can be confusing.

jkczyz commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -318,13 +318,13 @@ pub trait ChannelKeys : Send+Clone {
 	fn sign_channel_announcement<T: secp256k1::Signing>(&self, msg: &msgs::UnsignedChannelAnnouncement, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
 
-	/// Set the remote channel basepoints and remote/local to_self_delay.
+	/// Set the remote channel basepoints and counterparty/our local_to_self_delay.
```
Remove "our"?

ariard commented:
Updated with last comments.

TheBlueMatt commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -319,13 +319,13 @@ pub trait ChannelKeys : Send+Clone {
 	fn sign_channel_announcement<T: secp256k1::Signing>(&self, msg: &UnsignedChannelAnnouncement, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
 
-	/// Set the remote channel basepoints and remote/local to_self_delay.
+	/// Set the remote channel basepoints and counterparty/local_to_self_delay.
```
What do we think about `counterparty_set_to_local_delay` and `local_set_to_counterparty_delay`? I still have a hard time thinking that this is that much clearer.

TheBlueMatt commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -384,7 +384,7 @@ pub struct InMemoryChannelKeys {
 	/// Local public keys and basepoints
 	pub(crate) local_channel_pubkeys: ChannelPublicKeys,
-	/// Remote public keys and remote/local to_self_delay, populated on channel acceptance
+	/// Remote public keys and counterparty/local to_self_delay, populated on channel acceptance
```
nit: still says to_self_delay.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -479,31 +484,31 @@ pub(crate) fn get_htlc_redeemscript_with_explicit_keys(htlc: &HTLCOutputInCommit
 
 /// panics if htlc.transaction_output_index.is_none()!
-pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, to_self_delay: u16, htlc: &HTLCOutputInCommitment, a_delayed_payment_key: &PublicKey, revocation_key: &PublicKey) -> Transaction {
+pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, to_self_delay: u16, htlc: &HTLCOutputInCommitment, delayed_payment_key: &PublicKey, revocation_key: &PublicKey) -> Transaction {
```
nit: maybe change to contest_delay here too?

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -523,7 +528,7 @@ pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, to_self_del
 	let mut txouts: Vec<TxOut> = Vec::new();
 	txouts.push(TxOut {
-		script_pubkey: get_revokeable_redeemscript(revocation_key, to_self_delay, a_delayed_payment_key).to_v0_p2wsh(),
+		script_pubkey: get_revokeable_redeemscript(revocation_key, to_self_delay, delayed_payment_key).to_v0_p2wsh(),
```
same.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -268,36 +268,41 @@ pub fn derive_public_revocation_key<T: secp256k1::Verification>(secp_ctx: &Secp2
+	/// Countersignatory's HTLC Key
+	pub countersignatory_htlc_key: PublicKey,
+	/// Broadcaster's Payment Key (which isn't allowed to be spent from for some delay)
+	pub delayed_payment_key: PublicKey,
```
Should we leave the broadcaster_ prefix here?

ariard replied:
There is only one delayed_payment_key per-commitment transaction. Previously `revocation_key` didn't have a prefix. This is converging towards this nomenclature antecedent.

TheBlueMatt replied:
Sure, its the only one, but there's no harm in indicating who's it is, and it can only help readability.

TheBlueMatt commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -532,21 +532,21 @@ impl ChannelKeys for InMemoryChannelKeys {
 			};
-			let local_htlcpubkey = match chan_utils::derive_public_key(&secp_ctx, &per_commitment_point, &self.pubkeys().htlc_basepoint) {
-				Ok(local_htlcpubkey) => local_htlcpubkey,
+			let htlcpubkey = match chan_utils::derive_public_key(&secp_ctx, &per_commitment_point, &self.pubkeys().htlc_basepoint) {
```
nit: Can we leave the local_ prefix here?

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -268,36 +268,41 @@ pub fn derive_public_revocation_key<T: secp256k1::Verification>(secp_ctx: &Secp2
-	let mut part_b = per_commitment_point.clone();
-	part_b.mul_assign(&secp_ctx, &commit_append_rev_hash_key)?;
-	part_a.combine(&part_b)
+	let mut counterparty_contrib = revocation_base_point.clone();
```
Here, on the other hand, my point at https://github.com/rust-bitcoin/rust-lightning/pull/633/files#r475152575 stands (and maybe its nice that they use the same nomenclature)

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -268,36 +268,41 @@ pub fn derive_public_revocation_key<T: secp256k1::Verification>(secp_ctx: &Secp2
 /// The set of public keys which are used in the creation of one commitment transaction.
 /// These are derived from the channel base keys and per-commitment data.
 ///
+/// A broadcaster key is coming from potential broadcaster of the computed transaction.
```
s/is coming from/is provided by/

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -268,36 +268,41 @@ pub fn derive_public_revocation_key<T: secp256k1::Verification>(secp_ctx: &Secp2
 ///
+/// A broadcaster key is coming from potential broadcaster of the computed transaction.
+/// A countersignatory key is coming from a protocol participant unable to broadcast by its
+/// own the computed transaction.
```
s/broadcast by its own the computed transaction/to broadcast the transaction/.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -479,31 +484,31 @@ pub(crate) fn get_htlc_redeemscript_with_explicit_keys(htlc: &HTLCOutputInCommit
 }
 
-/// note here that 'a_revocation_key' is generated using b_revocation_basepoint and a's
+/// note here that 'revocation_key' is generated using remote_revocation_basepoint and broadcaster's
```
countersignatory, not remote

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -479,31 +484,31 @@ pub(crate) fn get_htlc_redeemscript_with_explicit_keys(htlc: &HTLCOutputInCommit
 
 /// panics if htlc.transaction_output_index.is_none()!
-pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, to_self_delay: u16, htlc: &HTLCOutputInCommitment, a_delayed_payment_key: &PublicKey, revocation_key: &PublicKey) -> Transaction {
+pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, contest_delay: u16, htlc: &HTLCOutputInCommitment, delayed_payment_key: &PublicKey, revocation_key: &PublicKey) -> Transaction {
```
broadcaster_delayed_payment_key, I believe?

ariard replied:
See point above. We should scope `revocation_key` or let it as it.

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -277,18 +277,18 @@ pub(super) struct Channel<ChanSigner: ChannelKeys> {
 
-	cur_local_commitment_transaction_number: u64,
-	cur_remote_commitment_transaction_number: u64,
+	cur_commitment_transaction_number: u64,
```
Can we keep local here?

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -326,19 +326,19 @@ pub(super) struct Channel<ChanSigner: ChannelKeys> {
 	holding_cell_update_fee: Option<u32>,
-	next_local_htlc_id: u64,
-	next_remote_htlc_id: u64,
+	next_htlc_id: u64,
```
same here.

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -442,7 +442,7 @@ macro_rules! secp_check {
 impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 	// Convert constants + channel value to limits:
-	fn get_our_max_htlc_value_in_flight_msat(channel_value_satoshis: u64) -> u64 {
+	fn get_max_htlc_value_in_flight_msat(channel_value_satoshis: u64) -> u64 {
```
Same here and for derive_dust_limit.

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -618,9 +618,9 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-		let max_to_self_delay = u16::min(config.peer_channel_config_limits.their_to_self_delay, MAX_LOCAL_BREAKDOWN_TIMEOUT);
-		if msg.to_self_delay > max_to_self_delay {
-			return Err(ChannelError::Close(format!("They wanted our payments to be delayed by a needlessly long period. Upper limit: {}. Actual: {}", max_to_self_delay, msg.to_self_delay)));
+		let max_locally_selected_contest_delay = u16::min(config.peer_channel_config_limits.their_to_self_delay, MAX_LOCAL_BREAKDOWN_TIMEOUT);
```
Shouldnt this be counteraprty_selected?

ariard replied:
I think it's a bit more subtle, it's the max counterparty_selected as enforced by our configuration, but yeah updated.

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -654,24 +654,24 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		let remote_channel_reserve_satoshis = Channel::<ChanSigner>::get_remote_channel_reserve_satoshis(msg.funding_satoshis);
-		if remote_channel_reserve_satoshis < our_dust_limit_satoshis {
-			return Err(ChannelError::Close(format!("Suitable channel reserve not found. remote_channel_reserve was ({}). our_dust_limit_satoshis is ({}).", remote_channel_reserve_satoshis, our_dust_limit_satoshis)));
+		if remote_channel_reserve_satoshis < dust_limit_satoshis {
```
Same, maybe don't drop our.

TheBlueMatt commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -80,7 +80,7 @@ pub enum SpendableOutputDescriptor {
 	/// script generation), you must pass the remote revocation_basepoint (which appears in the
 	/// call to ChannelKeys::on_accept) and the provided per_commitment point
-	/// to chan_utils::derive_public_revocation_key.
+	/// to chan_utils::derive_public_broadcaster_revocation_key.
```
The comments/variables/etc here still have a ton of references to remote_, to_self_delay, etc.

ariard replied:
Right corrected, note there is still remote/local references around SpendableOutputDescriptors as they're relative to ChannelMonitor (and I'll do post-anchor to avoid a confusing rebase).

TheBlueMatt commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -319,13 +319,13 @@ pub trait ChannelKeys : Send+Clone {
 	fn sign_channel_announcement<T: secp256k1::Signing>(&self, msg: &UnsignedChannelAnnouncement, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
 
-	/// Set the remote channel basepoints and remote/local to_self_delay.
+	/// Set the remote channel basepoints and counterparty_selected/locally_selected_contest_delay.
```
Should we change the function names in ChannelKeys (above)

ariard replied:
Add a new commit scoping ChannelKeys under new nomenclature.

TheBlueMatt commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -80,7 +80,7 @@ pub enum SpendableOutputDescriptor {
 	/// To derive the remote_revocation_pubkey provided here (which is used in the witness
```
This still says remote_, as does the variable name further down.

ariard replied:
Updated all ChannelMonitor/OnchainTxHandler-side, as it was tied down with using this version of Remote.

TheBlueMatt commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -499,66 +499,66 @@ impl ChannelKeys for InMemoryChannelKeys {
+		let channel_funding_redeemscript = make_funding_redeemscript(&funding_pubkey, &counterparty_channel_data.counterparty_channel_pubkeys.funding_pubkey);
 
-		Ok(local_commitment_tx.get_local_sig(&self.funding_key, &channel_funding_redeemscript, self.channel_value_satoshis, secp_ctx))
+		Ok(holder_commitment_tx.get_local_sig(&self.funding_key, &channel_funding_redeemscript, self.channel_value_satoshis, secp_ctx))
```
Maybe lets rename this to `get_commitment_sig` or `get_tx_sig` to mirror `get_htlc_sigs` instead of using the "local" name.

ariard replied:
Done, LocalCommitmentTx -> HolderCommitmentTx

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -217,87 +217,100 @@ pub fn derive_public_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, per_com
 /// Derives a per-commitment-transaction revocation key from its constituent parts.
 ///
+/// Only the transaction broadcaster owns a valid witness to propagate a revoked
+/// commitment transaction, thus per_commitment_secret always come from broadcaster
```
nit: s/come/comes/, twice plus in comments below.

ariard replied:
Not gonna to take it, to avoid modifying commits

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -217,87 +217,100 @@ pub fn derive_public_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, per_com
 
 /// Derives a per-commitment-transaction revocation key from its constituent parts.
 ///
+/// Only the transaction broadcaster owns a valid witness to propagate a revoked
```
s/revoked/revocable/ plus in the comment below.

ariard replied:
Not gonna to take it, to avoid modifying commits.

TheBlueMatt replied:
Wait, isn't this fully opposite, though - only the countersignatory of the originally-broadcasted commitment transaction can create a valid witness to broadcast a revocation key? I suppose its now a little confused - the commitment-tx-countersignatory is now the broadcaster of the revocation-punishment transaction (hence the per_commitment_secret is sent). I don't know that this needs to be captured in the names per se, but its probably worth at least fixing the comment (a new commit is fine by me, though).

jkczyz commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -319,13 +319,13 @@ pub trait ChannelKeys : Send+Clone {
 	fn sign_channel_announcement<T: secp256k1::Signing>(&self, msg: &UnsignedChannelAnnouncement, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
 
-	/// Set the remote channel basepoints and counterparty/local_to_self_delay.
+	/// Set the counterparty channel basepoints and counterparty_selected/locally_selected_contest_delay.
```
Any reason to prefer "locally" over "holder"?

ariard replied:
Add a cleanup commit, iteration from a previous round of review.

jkczyz reviewed (commented):
"holder" is growing on me. :)

ariard commented on lightning/src/ln/channel.rs:
```diff
@@ -277,18 +277,18 @@ pub(super) struct Channel<ChanSigner: ChannelKeys> {
 
 	#[cfg(not(test))]
-	local_keys: ChanSigner,
+	holder_keys: ChanSigner,
```
Add a solo comment in Channel. As it now used everywhere I don't know if it useful to replicate through the codebase, but not strong opinion ?

ariard commented:
Add 4 new commits at 5b16f01 to address last round of reviews :)

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -848,7 +851,7 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		let mut included_dust_htlcs: Vec<(HTLCOutputInCommitment, Option<&HTLCSource>)> = Vec::new();
 
-		let dust_limit_satoshis = if local { self.our_dust_limit_satoshis } else { self.their_dust_limit_satoshis };
+		let holder_dust_limit_satoshis = if local { self.holder_dust_limit_satoshis } else { self.counterparty_dust_limit_satoshis };
```
This is confusing - holder_dust_limit_satoshis may be counterparty_dust_limit_satoshis.

TheBlueMatt replied:
I believe this (and below) should be broadcaster.

ariard replied:
Fixed, and reference below too. FYI I've a commit simplifying code on my anchor branch : 0a9430d

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -965,15 +968,15 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			// channel_reserve *or* it is making progress towards it.
-			let mut max_commitment_tx_output = if generated_by_local {
-				self.max_commitment_tx_output_local.lock().unwrap()
+			let mut holder_max_commitment_tx_output = if generated_by_local {
```
Same here.

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -1371,18 +1374,18 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		}
 		if msg.dust_limit_satoshis > 21000000 * 100000000 {
-			return Err(ChannelError::Close(format!("Peer never wants payout outputs? dust_limit_satoshis was {}", msg.dust_limit_satoshis)));
+			return Err(ChannelError::Close(format!("Peer never wants payout outputs? holder_dust_limit_satoshis was {}", msg.dust_limit_satoshis)));
```
I dont think this line needs to change but its counterparty

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -1415,16 +1418,16 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		}
 		if msg.dust_limit_satoshis < config.peer_channel_config_limits.min_dust_limit_satoshis {
-			return Err(ChannelError::Close(format!("dust_limit_satoshis ({}) is less than the user specified limit ({})", msg.dust_limit_satoshis, config.peer_channel_config_limits.min_dust_limit_satoshis)));
+			return Err(ChannelError::Close(format!("holder_dust_limit_satoshis ({}) is less than the user specified limit ({})", msg.dust_limit_satoshis, config.peer_channel_config_limits.min_dust_limit_satoshis)));
```
Same here and below.

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -3695,23 +3697,23 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-				return Err(ChannelError::Ignore("Cannot send value that would put them under remote channel reserve value".to_owned()));
+			let counterparty_commit_tx_fee_msat = self.next_remote_commit_tx_fee_msat(1);
+			if counterparty_balance_msat < holder_selected_chan_reserve_msat + counterparty_commit_tx_fee_msat {
+				return Err(ChannelError::Ignore("Cannot send value that would put our balance under counterparty-announced channel reserve value".to_owned()));
```
Wrong way around, this is the counterparty balance.

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -1445,58 +1448,58 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-		let local_keys = self.build_local_transaction_keys(self.cur_local_commitment_transaction_number)?;
-		let local_initial_commitment_tx = self.build_commitment_transaction(self.cur_local_commitment_transaction_number, &local_keys, true, false, self.feerate_per_kw, logger).0;
-		let local_sighash = hash_to_message!(&bip143::SighashComponents::new(&local_initial_commitment_tx).sighash_all(&local_initial_commitment_tx.input[0], &funding_script, self.channel_value_satoshis)[..]);
+		let keys = self.build_local_transaction_keys(self.cur_holder_commitment_transaction_number)?;
```
Note that there are a ton of places in channel.rs that still have the /local_// conversion instead of /local_/holder_/. It seems they're all in function-scope variables so I'm ok with it if you dont want to fix it, but figured I'd note in case you or Jeff feel strongly.

ariard replied:
Yes at first try I was aiming to savagely converse everything then figured out a lot of references are meddled with ChannelState names.

I'm aiming to fix it when we refactor/cleanup further Channel internals. And have some on anchor branch.

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -3219,7 +3221,7 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 	/// Gets the fee we'd want to charge for adding an HTLC output to this Channel
 	/// Allowed in any state (including after shutdown)
-	pub fn get_our_fee_base_msat<F: Deref>(&self, fee_estimator: &F) -> u32
+	pub fn get_fee_base_msat<F: Deref>(&self, fee_estimator: &F) -> u32
```
This should probably be _holder_

TheBlueMatt commented on lightning/src/ln/channel.rs:
```diff
@@ -3136,13 +3138,13 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-	pub fn get_their_htlc_minimum_msat(&self) -> u64 {
-		self.our_htlc_minimum_msat
+	pub fn get_counterparty_htlc_minimum_msat(&self) -> u64 {
+		self.holder_htlc_minimum_msat
```
Not a new bug, but this looks very broken - I'm kinda sad we don't have test coverage for it.

ariard replied:
That's was the starting point of : https://github.com/rust-bitcoin/rust-lightning/pull/670, and we know how it ends up. Add a reference in #680

TheBlueMatt replied:
Wait, I dont recall you mentioning this - I was under the impression 670 was NOTABUG, this is definitely a bug.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -217,87 +217,100 @@ pub fn derive_public_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, per_com
+	/// The broadcaster's revocation key which is used to allow the broadcaster of the commitment
+	/// transaction to provide their counterparty the ability to punish them if they broadcast
+	/// an old state.
+	pub broadcaster_revocation_key: PublicKey,
```
I think I'd be happier if this just said "revocation_key" to match "per_commitment_point" - its a private key that requires private data from both sides, and really only the countersignatory can ever learn the private key part, but its awkward to suggest its owned by either side. Note that this change would effect a few other cases of "revocation_key" in this file.

ariard replied:
Good call, shared keys are hard to attribute ownership. Corrected and normally other references spread over.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -217,87 +217,100 @@ pub fn derive_public_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, per_com
-pub fn derive_private_revocation_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, per_commitment_secret: &SecretKey, revocation_base_secret: &SecretKey) -> Result<SecretKey, secp256k1::Error> {
-	let revocation_base_point = PublicKey::from_secret_key(&secp_ctx, &revocation_base_secret);
-	let per_commitment_point = PublicKey::from_secret_key(&secp_ctx, &per_commitment_secret);
+pub fn derive_private_broadcaster_revocation_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, broadcaster_per_commitment_secret: &SecretKey, countersignatory_revocation_base_secret: &SecretKey) -> Result<SecretKey, secp256k1::Error> {
```
Same here as below, I think we shouldn't refer to the revocation keys as "broadcaster" or "countersignatory" - they're not really "owned" by either side.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -358,28 +371,28 @@ impl_writeable!(ChannelPublicKeys, 33*5, {
-/// Encumbering a `to_local` output on a commitment transaction or 2nd-stage HTLC transactions.
-pub fn get_revokeable_redeemscript(revocation_key: &PublicKey, to_self_delay: u16, delayed_payment_key: &PublicKey) -> Script {
+/// key or the broadcaster_delayed_payment_key and satisfying the relative-locktime OP_CSV constrain.
+/// Encumbering a `to_holder` output on a commitment transaction or 2nd-stage HTLC transactions.
```
This should still read to_local - it may actually be to our counterparty.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -479,31 +492,31 @@ pub(crate) fn get_htlc_redeemscript_with_explicit_keys(htlc: &HTLCOutputInCommit
 }
 
-/// note here that 'a_revocation_key' is generated using b_revocation_basepoint and a's
+/// note here that 'broadcaster_revocation_key' is generated using countersignatory_revocation_basepoint and broadcaster's
```
Note your fault but this comment is just stale - Probably replace it with "Gets the witness redeemscript for an HTLC output in a commitment transaction. Note that `htlc` does *not* need to have its previous_output_index filled."

ariard replied:
Added a commit.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -624,7 +637,7 @@ impl LocalCommitmentTransaction {
 			unsigned_tx,
 			their_sig,
```
This should be counterparty, no?

ariard replied:
Yes!

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -587,24 +600,24 @@ impl LocalCommitmentTransaction {
 	/// The unsigned transaction outputs must be consistent with htlc_data.  This function
 	/// only checks that the shape and amounts are consistent, but does not check the scriptPubkey.
-	pub fn new_missing_local_sig(unsigned_tx: Transaction, their_sig: Signature, our_funding_key: &PublicKey, their_funding_key: &PublicKey, local_keys: TxCreationKeys, feerate_per_kw: u32, htlc_data: Vec<(HTLCOutputInCommitment, Option<Signature>)>) -> LocalCommitmentTransaction {
+	pub fn new_missing_holder_sig(unsigned_tx: Transaction, their_sig: Signature, our_funding_key: &PublicKey, their_funding_key: &PublicKey, holder_keys: TxCreationKeys, feerate_per_kw: u32, htlc_data: Vec<(HTLCOutputInCommitment, Option<Signature>)>) -> HolderCommitmentTransaction {
```
Several variables in this line need their/our swapped for the new names.

ariard replied:
Yes updated a bunch

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -681,16 +694,16 @@ impl LocalCommitmentTransaction {
 	/// considered dust and not included, a None entry exists, for all others a signature is
 	/// included.
-	pub fn get_htlc_sigs<T: secp256k1::Signing + secp256k1::Verification>(&self, htlc_base_key: &SecretKey, local_csv: u16, secp_ctx: &Secp256k1<T>) -> Result<Vec<Option<Signature>>, ()> {
+	pub fn get_htlc_sigs<T: secp256k1::Signing + secp256k1::Verification>(&self, htlc_base_key: &SecretKey, holder_csv: u16, secp_ctx: &Secp256k1<T>) -> Result<Vec<Option<Signature>>, ()> {
```
Shouldn't this be changed to contest delay?

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -587,24 +600,24 @@ impl LocalCommitmentTransaction {
 			their_sig: dummy_sig,
 			our_sig_first: false,
-			local_keys: TxCreationKeys {
+			holder_keys: TxCreationKeys {
```
Huh, should this just be `keys` instead? Its awkward to have a holder_keys.countersignatory_htlc_keys field.

TheBlueMatt commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -101,8 +101,8 @@ pub enum SpendableOutputDescriptor {
-		/// The remote_revocation_pubkey used to derive witnessScript
-		remote_revocation_pubkey: PublicKey
+		/// The counterparty_revocation_pubkey used to derive witnessScript
+		counterparty_revocation_pubkey: PublicKey
```
s/counterparty//

TheBlueMatt commented on lightning/src/chain/keysinterface.rs:
```diff
@@ -71,14 +71,14 @@ pub enum SpendableOutputDescriptor {
 	///
-	/// To derive the remote_revocation_pubkey provided here (which is used in the witness
-	/// script generation), you must pass the remote revocation_basepoint (which appears in the
+	/// To derive the counterparty_revocation_pubkey provided here (which is used in the witness
```
s/counterparty// (but not on the next line down).

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -335,7 +350,7 @@ pub struct ChannelPublicKeys {
 	/// states.
 	pub revocation_basepoint: PublicKey,
 	/// The public key which receives our immediately spendable primary channel balance in
-	/// remote-broadcasted commitment transactions. This key is static across every commitment
```
Not your bug, but can you fix this comment to not refer to ours/theirs/counterparty/etc - this struct is generic across both sides (and thus should only ever use broadcaster/countersignatory).

ariard replied:
Hmmm I keep the counterparty's it makese sense here, as this immediate spend is only available if it's the counterparty commitment ?

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -415,17 +430,17 @@ impl_writeable!(HTLCOutputInCommitment, 1 + 8 + 4 + 32 + 5, {
 		Builder::new().push_opcode(opcodes::all::OP_DUP)
 		              .push_opcode(opcodes::all::OP_HASH160)
-		              .push_slice(&PubkeyHash::hash(&revocation_key.serialize())[..])
+		              .push_slice(&PubkeyHash::hash(&broadcaster_revocation_key.serialize())[..])
```
There are still a number of references to broadcaster_revocation_key - can you remove those too?

TheBlueMatt reviewed (commented):
Still reviewing, but one comment came up.

ariard commented on lightning/src/ln/channel.rs:
```diff
@@ -1856,13 +1859,12 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
-			if self.value_to_self_msat < self.local_channel_reserve_satoshis * 1000 + local_commit_tx_fee_msat {
-				return Err(ChannelError::Close("Cannot receive value that would put us under local channel reserve value".to_owned()));
+			if self.value_to_self_msat < self.counterparty_selected_channel_reserve_satoshis * 1000 + local_commit_tx_fee_msat {
+				return Err(ChannelError::Close("Cannot accept HTLC that would put counterparty balance under our-announced channel reserve value".to_owned()));
```
Corrected.

TheBlueMatt commented on lightning/src/ln/chan_utils.rs:
```diff
@@ -334,8 +349,8 @@ pub struct ChannelPublicKeys {
-	/// The public key which receives our immediately spendable primary channel balance in
-	/// remote-broadcasted commitment transactions. This key is static across every commitment
+	/// The public key which receives an immediately spendable primary channel balance in
+	/// a broadcaster's commitment transactions. This key is static across every commitment
```
This reads a bit confusing - its the countersignatory's balance in a broadcaster's transaction, but reading it I'd think its a broadcaster's balance.

TheBlueMatt reviewed (commented):
One more note plus this needs squashed or at least 4491986f12aa83b30abc3f87892473bea274ab52 fixed (it doesn't build on its own).

TheBlueMatt commented:
Verified that every commit except 4491986f12aa83b30abc3f87892473bea274ab52 builds + passes tests, so just that one commit needs fixedup/the next squashed into it.
